packages/blade/src/components/List/__tests__/List.native.test.tsx

```diff
@@ -0,0 +1,181 @@
+        <ListItem>Level 2</ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render unordered List of small size', () => {
+    const { toJSON } = renderWithTheme(
+      <List variant="unordered" size="small">
+        <ListItem>
+          Level 1
+          <List>
+            <ListItem>
+              Level 2
+              <List>
+                <ListItem>Level 3</ListItem>
+              </List>
+            </ListItem>
+          </List>
+        </ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render small unordered List with icon', () => {
+    const { toJSON } = renderWithTheme(
+      <List variant="unordered" size="small" icon={ArrowRightIcon}>
+        <ListItem>
+          Level 1
+          <List>
+            <ListItem icon={ArrowUpIcon}>
+              Level 2
+              <List>
+                <ListItem>Level 3</ListItem>
+              </List>
+            </ListItem>
+          </List>
+        </ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render medium unordered List with icon', () => {
+    const { toJSON } = renderWithTheme(
+      <List variant="unordered" size="medium" icon={ArrowRightIcon}>
+        <ListItem>
+          Level 1
+          <List>
+            <ListItem icon={ArrowUpIcon}>
+              Level 2
+              <List>
+                <ListItem>Level 3</ListItem>
+              </List>
+            </ListItem>
+          </List>
+        </ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render small ordered List', () => {
+    const { toJSON } = renderWithTheme(
+      <List variant="ordered" size="small">
+        <ListItem>
+          Level 1
+          <List>
+            <ListItem>
+              Level 2
+              <List>
+                <ListItem>Level 3</ListItem>
+              </List>
+            </ListItem>
+          </List>
+        </ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render medium ordered List', () => {
+    const { toJSON } = renderWithTheme(
+      <List variant="ordered" size="medium">
+        <ListItem>
+          Level 1
+          <List>
+            <ListItem>
+              Level 2
+              <List>
+                <ListItem>Level 3</ListItem>
+              </List>
+            </ListItem>
+          </List>
+        </ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render small ordered-filled List', () => {
+    const { toJSON } = renderWithTheme(
+      <List variant="ordered-filled" size="small">
+        <ListItem>Level 1</ListItem>
+        <ListItem>Level 2</ListItem>
+        <ListItem>Level 3</ListItem>
+      </List>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render medium ordered-filled List', () => {
```

Comment:
add tests for error cases?
* What happens when you pass some other element apart from ListItem to List?

Reply:
Added

Reply:
Added